spyder/plugins/completion/plugin.py

```diff
@@ -170,19 +189,36 @@ def gather_and_send(self, principal_source, response_instance, req_type,
             # removed before the response can be processed.
             pass
 
-    def send_request(self, language, req_type, req, req_id=None):
+    def _is_client_running(self, name):
+        return self.clients.get(name, {}).get('status', self.STOPPED) == self.RUNNING
+
+    def send_request(self, language, req_type, req):
         req_id = self.req_id
-        for client_name in self.clients:
+
+        client_names = []
+        if req_type not in self.MUTEX_REQUEST_TYPES:
```

Comment:
Don't we need any language-related check here?

Reply:
We didn't have any such checks before, so I assume they weren't necessary before.

Kite returns an empty response for unsupported languages (updated in this PR), and LSP throws away requests for unsupported languages.

Since we try Kite first, and it will always return a response for all languages, the LSP fallback will always get triggered, even if Kite is first called on an unsupported language, so this should all work.

That being said, I'm open to other suggestions for how to structure this.